promql/functions.go

```diff
@@ -830,6 +831,44 @@ func funcLabelReplace(ev *evaluator, args Expressions) model.Value {
 	return vector
 }
 
+// === label_join(vector model.ValVector, dest_labelname, separator, src_labelname...) Vector ===
+func funcLabelJoin(ev *evaluator, args Expressions) model.Value {
+	var (
+		vector    = ev.evalVector(args[0])
+		dst       = model.LabelName(ev.evalString(args[1]).Value)
+		sep       = ev.evalString(args[2]).Value
+		srcLabels = make([]model.LabelName, 0)
+	)
+	for i := 3; i < len(args); i++ {
+		src := args[i]
+		srcLabels = append(srcLabels, model.LabelName(ev.evalString(src).Value))
+	}
+
+	if !model.LabelNameRE.MatchString(string(dst)) {
```

Comment:
We should validate the `srcLabels` as well. That will help people catch typos in their queries.